### ISSUE TITLE
Affinity Publisher errors with Junicode VF

### ISSUE THREAD

**Issue by morbus** [psb1558/Junicode-font#302]
The Affinity series of products added variable font support in the 2.5.x series. I've been attempting to use Junicode VF with them, but have run into a number of issues (and I've been able to reproduce these using the non-VF TTFs as well). I DO NOT consider these problems with Junicode itself (as I'm able to use other apps with Junicode VF with no known issue), but I did just want to provide the Affinity bug reports as an FYI:

1. https://forum.affinity.serif.com/index.php?/topic/219433-junicode-font-crashes-pdf-export/
2. https://forum.affinity.serif.com/index.php?/topic/219401-many-glyphed-fonts-slow-down-text-entry-and-dragging/

Close this issue with impunity.

**Comment by morbus** [comment]
Although, in both of my test cases above, I DID use the Junicode VF Bold Italic font, and there's mention that there are issues with the Italics font in Indesign here: https://github.com/psb1558/Junicode-font/issues/220. So... maybe it's all related?

**Comment by kenmcd** [comment]
Peter, 
This seems to be issues with APub, not Junicode.
I removed all the OpenType features from the v2.209 Roman VF.
Then the export to PDF did not crash.

Export to PDF is always slow with APub and larger variable fonts as they seem to process every glyph in the entire font.
SF Pro Variable can take 10-15 minutes to export to PDF.
Same amount of time to just show the preview.
It has 27,262 glyphs, and 30 masters.
APub's little brain just grinds and grinds.
The font processing needs to be optimized.

Don't think this is anything wrong in Junicode.

**Comment by psb1558** [comment]
The great news here is Affinity supporting variable fonts. I believe I read not long ago that this wasn't on their radar at all: they must have gotten significant pressure from users.

I have found it a hit-and-miss business to work around application bugs triggered by my fonts' features, though with the help of folks like @kenmcd I have found a few of them. But the best policy for Junicode, I think, is to make sure it is standards-compliant. Then, if it doesn't work in one app or another, it is on the maker of the app to implement the fix.

I have never seen a rollout of VF support that was problem-free. Affinity seems reasonably good at responding to user input and fixing bugs: I'm sure they'll get it right sooner or later.
